internal/circuit/redis.go

```
@@ -0,0 +1,193 @@
+// RecordTerminalFailure records one failure and returns the new state.
+func (s *RedisStore) RecordTerminalFailure(ctx context.Context, provider string) (State, error) {
+	now := float64(time.Now().UnixNano()) / 1e9
+	cutoff := now - float64(s.cfg.WindowSeconds)
+
+	res, err := luaRecordFailure.Run(ctx, s.rdb,
+		[]string{s.failuresKey(provider), s.stateKey(provider)},
+		now, cutoff, s.cfg.FailureThreshold, s.cfg.CooldownSeconds,
+	).Text()
+	if err != nil {
+		// On Redis error fail safe: return closed so we don't block traffic.
+		return StateClosed, nil
```

golangci-lint (2.11.4)

[error] error is not nil (line 123) but it returns nil

(nilerr)